### PR TITLE
Explain the goals of Jakarta Data and motivate/justify/explain the selected design

### PR DESCRIPTION
**Pull request by otaviojava**
# Changes

- Enhance documentation
- Explain what DDD is
- Compare with other patterns

Solves: https://github.com/jakartaee/data/issues/222

### CODE REVIEW

**Review by mpredli01 — APPROVED** [review]
LGTM

**Review by mpredli01 — APPROVED** [review]
LGTM

**Review by mpredli01 — APPROVED** [review]
LGTM

**Review by mpredli01 — COMMENTED** [review]
I don't think this is a big deal, but perhaps I should mention it:  The use of "portable" could be somewhat misconstrued since: "'agnostic' does not mean 'portable'" is mentioned at the beginning, then followed later on with "This approach ensures that your queries and operations are more portable."  What do you think?

**Review by mpredli01 — APPROVED** [review]
LGTM

**Comment by otaviojava** [comment]
@mpredli01 ok, I will change this point.

**Comment by otaviojava** [comment]
@mpredli01 what do you think?

File: spec/src/main/asciidoc/chapters/introduction/introduction.asciidoc [hunk 1]
```diff
@@ -25,17 +25,37 @@ Dealing with a database is one of the biggest challenges within a software archi
...
 
-* Be a persistence agnostic API. Therefore, through abstractions, it will connect different types of databases and storage sources.
-* Be a pluggable and extensible API. Even when the API won't support a particular behavior of a storage engine, it might provide an extensible API to make it possible.
+The primary problem Jakarta Data sets out to solve is the complexity and inconsistency that arises when Java applications encounter various database systems—some relational, some NoSQL, and others unique. Managing these diverse data sources can be daunting, often requiring developers to write specialized code for each storage technology.
+
+Jakarta Data takes this challenge further by enhancing the concept of a "persistence agnostic API" with a Domain-Driven Design (DDD) approach. This approach enables developers to work with different databases and storage engines and align their data access strategies with the core principles of DDD, where the domain model plays a central role in shaping the application's architecture.
```
Comment (njr-11) [inline spec/src/main/asciidoc/chapters/introduction/introduction.asciidoc:32]:
```suggestion
Jakarta Data combines the concept of a persistence agnostic API with a Domain-Driven Design (DDD) approach. This approach enables developers to work with different databases and storage engines and align their data access strategies with the core principles of DDD, where the domain model plays a central role in shaping the application's architecture.
```

File: spec/src/main/asciidoc/chapters/introduction/introduction.asciidoc [hunk 2]
```diff
@@ -25,17 +25,37 @@ Dealing with a database is one of the biggest challenges within a software archi
...
+
+Jakarta Data is guided by a set of clear and well-defined objectives to simplify data integration and enhance data access for Java developers. These objectives serve as the pillars of its design philosophy, ensuring that it addresses real-world challenges and provides concrete advantages to developers:
+
+* **Jakarta Data is engineered to tackle a fundamental problem:** simplifying data access and manipulation within Java applications that interact with diverse databases and storage sources.
+
+* **Jakarta Data is designed to be "persistence agnostic."** In this context, "agnostic" does not mean "transferable" but implies that Jakarta Data is not tied to a specific database technology. It offers a flexible, adaptable framework that allows you to work with the databases and storage sources that best suit your project's needs. This agnostic approach ensures that Jakarta Data can cater to various use cases.
```
Comment (njr-11) [inline spec/src/main/asciidoc/chapters/introduction/introduction.asciidoc:38]:
```suggestion
* **Jakarta Data is designed to be persistence agnostic.** In this context, agnostic does not mean that you can switch the underlying persistence without changes but implies that Jakarta Data is not tied to a specific database technology. It offers a flexible, adaptable framework that allows you to work with the databases and storage sources that best suit your project's needs. This agnostic approach ensures that Jakarta Data can cater to various use cases.
```

File: spec/src/main/asciidoc/repository.asciidoc [hunk 3]
```diff
@@ -18,13 +18,25 @@ In Domain-Driven Design (DDD) the repository pattern encapsulates the logic requ
...
+
+- **Active Record Pattern:** In the Active Record pattern, domain objects are tightly coupled with database operations. Each domain object typically represents a database record, and the object itself handles database CRUD (Create, Read, Update, Delete) operations. It can be simpler for small-scale projects but can lead to a lack of separation of concerns in larger, more complex applications.
+
+- **DAO (Data Access Object) Pattern:** The DAO pattern separates the data access code from domain objects. DAO classes handle database interactions, while domain objects focus solely on representing data. While this separation provides better maintainability and testability, it can lead to more boilerplate code than the Repository pattern.
+
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:38]:
```suggestion
```
This section seemly oversimplified and would really need to go much more in depth to serve as a fair comparison of advantages and trade-offs of different approaches.  I recommend removing this section and only focusing on the approach that is offered by Jakarta Data.

**Review by njr-11 — APPROVED** [review]
Regarding the comment that was made about portable/agnostic.  I don't think the word transferable is any better than portable.  I added a few suggestions in that area to hopefully improve clarity, which you can include if you think they help, otherwise feel free to ignore them.

**Comment by otaviojava** [comment]
@njr-11 thank you for all the suggestions.
@mpredli01 you too.

**Pull request merged**